app/src/components/ChatBot/chat-bot.tsx

```diff
@@ -58,20 +66,251 @@ export default function ChatBot({
+          error,
+          "Failed to initialize chat. Please refresh the page.",
+        );
+      }
+    };
+    if (!threadId) {
+      initializeThread();
+    }
+  }, []); // Empty dependency array means this effect runs only once
+
+  // Automatically scroll to bottom of chat
+  const messagesEndRef = useRef<HTMLDivElement | null>(null);
+  const scrollToBottom = () => {
+    messagesEndRef.current?.scrollIntoView({ behavior: "smooth" });
+  };
+  useEffect(() => {
+    scrollToBottom();
+  }, [messages]);
+
+  // TODO: Convert to Redux
+  const sendMessage = async (text: string) => {
+    try {
+      const response = await fetch(`/api/v0/assistants/threads/messages`, {
+        method: "POST",
+        body: JSON.stringify({
+          threadId: threadId,
+          content: text,
+        }),
+      });
+
+      if (!response.ok) {
+        throw new Error(`HTTP error! status: ${response.status}`);
+      }
+      if (response.body == null) {
+        throw new Error("Response body is null");
+      }
+
+      const stream = AssistantStream.fromReadableStream(response.body);
+      handleReadableStream(stream);
+    } catch (error) {
+      handleError(error, "Failed to send message. Please try again.");
+      setInputDisabled(false);
+    }
+  };
+
+  ////////////////////
+  // Function calls //
+  ////////////////////
+
+  const functionCallHandler = async (call: any) => {
+    try {
+      // Handle function get all data sources
+      if (call?.function?.name === "get_all_datasources") {
+        const { data, error } = await getAllDataSources({
+          inventoryId,
+        });
+        if (error) throw error;
+        return JSON.stringify(data);
+
+        // Handle function to get all user inventories
+      } else if (call?.function?.name === "get_user_inventories") {
+        const { data, error } = await getUserInventories();
+        if (error) throw error;
+        return JSON.stringify(data);
+
+        // Handle function to get details of specific inventory
+      } else if (call?.function?.name === "get_inventory") {
+        // Parse the nested JSON string in the "arguments" field
+        const argument = JSON.parse(call.function.arguments);
+        const selectedInventoryId = argument.inventory_id;
+        const { data, error } = await getInventory(selectedInventoryId);
+        if (error) throw error;
+        return JSON.stringify(data);
+      }
+      // Handle if no function call was identified
+      else {
+        throw new Error("No function identified to call");
+      }
+    } catch (error) {
+      handleError(error, `Error in function call: ${call?.function?.name}`);
+      return JSON.stringify({
+        error: { error: error, message: "Error in function call" },
+      });
+    }
+  };
+
+  const submitActionResult = async (
+    threadId: string,
+    runId: string,
+    toolCallOutputs: object,
+  ) => {
+    try {
+      const response = await fetch(`/api/v0/assistants/threads/actions`, {
```

Comment:
Please use the a Redux Toolkit API hook (defined in src/services/api/ts) to fetch API data.

Reply:
Created a new ticket and added number ON-2137 to todo